src/main/java/se/liu/ida/hefquin/engine/queryplan/utils/TextBasedLogicalPlanPrinterImpl2.java

```diff
@@ -0,0 +1,62 @@
+package se.liu.ida.hefquin.engine.queryplan.utils;
+
+import java.io.PrintStream;
+
+import se.liu.ida.hefquin.engine.queryplan.logical.LogicalPlan;
+
+public class TextBasedLogicalPlanPrinterImpl2 implements LogicalPlanPrinter
+{
+	// The string represents '|  '.
+	private static String levelIndentBase = "\u2502  ";
+	// The string represents '├──'.
+	private static String nonLastChildIndentBase = "\u251C\u2500\u2500";
+	// The string represents '└──'.
+	private static String lastChildIndentBase = "\u2514\u2500\u2500";
+		
+	@Override
+	public void print( final LogicalPlan plan, final PrintStream out ) {
+		planWalk(plan, 0, 0, 0, out);
+		out.flush();	
+	}
+	
+	public String getIndentLevelString(final int planNumber, final int planLevel, final int numberOfSiblings) {
+		String indentLevelString = "";
+		for ( int i = 1; i < planLevel; i++ ) {
+			indentLevelString += levelIndentBase;
+		}
+		if (planNumber < numberOfSiblings-1) {
+			indentLevelString += nonLastChildIndentBase ;
+		}
+		else if (numberOfSiblings > 0) {
+			indentLevelString += lastChildIndentBase;
+		}
+		else {
+			// This is only for the root operator.
+			indentLevelString = "";
+		}
+		return indentLevelString;
```

Comment:
Since the case of `numberOfSiblings == 0` is a special case (only for the root operator), I would handle it outside of the regular code path that is used for the subplans. This is to improve readability.
Additionally, I would include an additional check to make sure the special case is indeed invoked only for the root operator.
In summary, I would change the code of this function as follows.
```suggestion
		if ( numberOfSiblings == 0 ) {
			// This is only for the root operator of the overall plan to be printed.
			assert planLevel == 0;
			return "";
		}

		String indentLevelString = "";
		for ( int i = 1; i < planLevel; i++ ) {
			indentLevelString += levelIndentBase;
		}

		if ( planNumber < numberOfSiblings-1 ) {
			return indentLevelString + nonLastChildIndentBase ;
		}
		else {
			return indentLevelString + lastChildIndentBase;
		}
```